crates/iota/src/client_ptb/ptb.rs

```diff
@@ -205,22 +212,22 @@ impl PTB {
             }
         };
 
-        if program_metadata.json_set {
-            let json_string = if program_metadata.summary_set {
+        let result_string = if program_metadata.json_set {
+            if program_metadata.summary_set {
                 serde_json::to_string_pretty(&serde_json::json!(summary))
                     .map_err(|_| anyhow!("Cannot serialize PTB result to json"))?
             } else {
                 serde_json::to_string_pretty(&serde_json::json!(transaction_response))
                     .map_err(|_| anyhow!("Cannot serialize PTB result to json"))?
-            };
-            println!("{}", json_string);
+            }
         } else if program_metadata.summary_set {
-            println!("{}", Pretty(&summary));
+            Pretty(&summary).to_string()
         } else {
-            println!("{}", transaction_response);
-        }
+            transaction_response.to_string()
+        };
+        println!("{result_string}");
```

Comment:
I think it would be useful to have a function that creates the string without printing it, more flexible

Reply:
I tried this, but then I don't see a way to render the output in special ways like the help output does currently with some words for example **fat** and with __underline__

Reply:
Are you saying we are losing this information when converted to a string?

Reply:
Yes, unless we return StyledStr, but that seems to be more annoying to work with